server/src/main/kotlin/net/horizonsend/ion/server/features/starship/subsystem/weapon/projectile/ArsenalRocketProjectile.kt

```diff
@@ -0,0 +1,191 @@
+package net.horizonsend.ion.server.features.starship.subsystem.weapon.projectile
+
+import com.mojang.math.Transformation
+import net.horizonsend.ion.common.extensions.userError
+import net.horizonsend.ion.common.utils.miscellaneous.randomInt
+import net.horizonsend.ion.server.configuration.StarshipWeapons
+import net.horizonsend.ion.server.features.customitems.CustomItems
+import net.horizonsend.ion.server.features.starship.active.ActiveStarship
+import net.horizonsend.ion.server.features.starship.damager.Damager
+import net.horizonsend.ion.server.features.starship.subsystem.weapon.interfaces.AmmoConsumingWeaponSubsystem
+import net.horizonsend.ion.server.miscellaneous.utils.minecraft
+import net.horizonsend.ion.server.miscellaneous.utils.toBlockPos
+import net.minecraft.network.protocol.game.ClientboundAddEntityPacket
+import net.minecraft.world.entity.Display.ItemDisplay
+import net.minecraft.world.entity.EntityType
+import org.bukkit.Bukkit
+import org.bukkit.Color
+import org.bukkit.FluidCollisionMode
+import org.bukkit.Location
+import org.bukkit.Particle
+import org.bukkit.Particle.DustOptions
+import org.bukkit.block.Block
+import org.bukkit.craftbukkit.v1_20_R3.entity.CraftPlayer
+import org.bukkit.craftbukkit.v1_20_R3.inventory.CraftItemStack
+import org.bukkit.entity.Entity
+import org.bukkit.entity.Player
+import org.bukkit.inventory.ItemStack
+import org.bukkit.util.RayTraceResult
+import org.bukkit.util.Vector
+import org.joml.Quaternionf
+import org.joml.Vector3f
+
+class ArsenalRocketProjectile(
+	starship: ActiveStarship?,
+	loc: Location,
+	dir: Vector,
+	shooter: Damager,
+	var upOrDown: Boolean //Up = true, down = false
+) : SimpleProjectile(starship, loc, dir, shooter), DisplayEntityProjectile {
+	override val balancing: StarshipWeapons.ProjectileBalancing = starship?.balancing?.weapons?.arsenalMissile!!
+	override val range: Double = balancing.range
+	override val speed: Double = balancing.speed
+	override val starshipShieldDamageMultiplier = balancing.starshipShieldDamageMultiplier
+	override val areaShieldDamageMultiplier: Double = balancing.areaShieldDamageMultiplier
+	override val explosionPower: Float = balancing.explosionPower
+	override val volume: Int = balancing.volume
+	override val pitch: Float = balancing.pitch
+	override val soundName: String = balancing.soundName
+	override var displayEntities: MutableMap<Player, ItemDisplay?> = mutableMapOf()
+	override val originLocation: Location = loc
+	override var oldLocation: Location = originLocation
+	private var age: Int = 0 //Ticks
+	private var initialVelocity = this.dir
+	private var hasSwitchedToNormalDirection = false
+	private var targetedPosition: Location? = null
+	override var scale: Vector3f = Vector3f(3f,3f,3f)
+	override fun tick() {
+		delta = (System.nanoTime() - lastTick) / 1_000_000_000.0 // Convert to seconds
+		age++
+		if (age< randomInt(10,15) && !hasSwitchedToNormalDirection){
+			if (upOrDown) dir = Vector(0,1,0).multiply(dir.length())
```

Comment:
Cant this just be done by providing a blockface and using BlockFace#direction

Reply:
yeah I couldve, ill fix it